File: permissionsdispatcher-processor/src/main/java/permissions/dispatcher/processor/ClassType.java
package permissions.dispatcher.processor;

enum ClassType {
    ACTIVITY("target"),
    FRAGMENT("target.getActivity()");

    private final String activity;

    ClassType(String activity) {
        this.activity = activity;
    }

    static ClassType getClassType(String className, TypeResolver resolver) {
        if (resolver.isSubTypeOf(className, ConstantsProvider.ACTIVITY)) {
            return ACTIVITY;
        } else if (resolver.isSubTypeOf(className, ConstantsProvider.V4FRAGMENT)) {
            return FRAGMENT;
        }
        return null;
    }

    public String getActivity() {
        return activity;
    }

}

File: permissionsdispatcher-processor/src/main/kotlin/permissions/dispatcher/processor/ClassType.kt
package permissions.dispatcher.processor

enum class ClassType(public val activity: String) {
    ACTIVITY("target"),
    FRAGMENT("target.getActivity()");

    companion object {

        fun getClassType(className: String): ClassType? {
            if (className.endsWith("Activity")) {
                return ACTIVITY
            } else if (className.endsWith("Fragment")) {
                return FRAGMENT
            }
            return null
        }
    }

}
